### PR TITLE
check "max_number_of_predictions" parameter when testing

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- closes https://github.com/RasaHQ/rasa/issues/5489
- Check whether the circuit breaker would have tripped because `max_number_of_predictions` was exceeded when testing stories

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/processor.py [hunk 1]
```diff
@@ -500,19 +500,21 @@ def _should_handle_message(tracker: DialogueStateTracker):
             or tracker.latest_message.intent.get("name") == USER_INTENT_RESTART
         )
 
+    def is_action_limit_reached(
+        self, num_predicted_actions: int, should_predict_another_action: bool
+    ):
```
Comment (federicotdn) [inline rasa/core/processor.py:505]: We should add a docstring for this function.

File: rasa/core/constants.py [hunk 2]
```diff
@@ -65,3 +65,5 @@
 
 # RabbitMQ message property header added to events published using `rasa export`
 RASA_EXPORT_PROCESS_ID_HEADER_NAME = "rasa-export-process-id"
+
+CIRCUIT_BREAKER_TRIPPED = "circuit breaker tripped"
```
Comment (federicotdn) [inline rasa/core/constants.py:69]: I'm not sure if this is a proper constant, or a user-facing string. I can see you have only used it in one place, and it appears it eventually will be shown to the user along other action names. In this case I thing it's better to treat it as a regular string, that would imply removing this variable and putting the string directly in `_collect_action_executed_predictions`.

**Comment by melindaloubser1** [comment]
@federicotdn for the test, what would be a way to check if the circuit breaker trips without training a model? Since the default agent will always predict `action_listen` that won't work no matter how low the max_number_of_predictions is set. Should I just include training a model in the test?

File: rasa/core/processor.py [hunk 3]
```diff
@@ -500,19 +500,22 @@ def _should_handle_message(tracker: DialogueStateTracker):
...
```
Comment (federicotdn) [inline rasa/core/processor.py:505]:
```suggestion
    ) -> bool:
```
(apologies, forgot about this)

**Comment by federicotdn** [comment]
If the only way is to train a model, I think we should try that and see if we can build a useful test with it. I think the training data itself could go in the `data/` directory at the root of the project, probably in a subdirectory of its own.

File: rasa/core/processor.py (hunk 3 above)
Comment (melindaloubser1) [reply rasa/core/processor.py:505]: np

File: tests/core/test_evaluation.py [hunk 4]
```diff
@@ -82,6 +85,7 @@
 
 
 async def test_end_to_evaluation_with_forms(form_bot_agent: Agent):
+
```
Comment (federicotdn) [inline tests/core/test_evaluation.py:88]:
```suggestion
```

File: tests/core/test_evaluation.py [hunk 5]
```diff
@@ -91,3 +95,40 @@
     )
 
     assert not story_evaluation.evaluation_store.has_prediction_target_mismatch()
+
+
+async def test_end_to_evaluation_trips_circuit_breaker():
+
```
Comment (federicotdn) [inline tests/core/test_evaluation.py:101]:
```suggestion
```

File: data/test_evaluations/end_to_end_trips_circuit_breaker.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [story_trips_circuit_breaker](#story_trips_circuit_breaker)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
```
Comment (federicotdn) [inline data/test_evaluations/end_to_end_trips_circuit_breaker.md:8]: Looks like `doctoc` added some table of contents to your stories 😄  We should remove them, and from the other file as well. Maybe they got added by the pre-commit hook?

File: rasa/core/processor.py [hunk 7]
```diff
@@ -500,19 +500,22 @@ def _should_handle_message(tracker: DialogueStateTracker):
             or tracker.latest_message.intent.get("name") == USER_INTENT_RESTART
         )
 
+    def is_action_limit_reached(
+        self, num_predicted_actions: int, should_predict_another_action: bool
+    ) -> bool:
+        """Check whether the maximum number of predictions has been met"""
```
Comment (federicotdn) [inline rasa/core/processor.py:506]:
```suggestion
        """Check whether the maximum number of predictions has been met.
       
        Args:
            num_predictes_actions: Number of predicted actions.
            should_predict_another_action: Whether the last executed action allows
            for more actions to be predicted or not.
            
        Returns:
            `True` if the limit of actions to predict has been reached.
        """
```

File: data/test_evaluations/end_to_end_trips_circuit_breaker.md (hunk 6 above)
Comment (melindaloubser1) [reply data/test_evaluations/end_to_end_trips_circuit_breaker.md:8]: Oh boy! Yeah,  I did get some odd warnings about  the pre-commit hook

File: rasa/core/processor.py (hunk 7 above)
Comment (melindaloubser1) [reply rasa/core/processor.py:506]: Ah, sorry, was not sure how complete of docstrings I should put

**Pull request merged**